source/addonStore/models/addon.py

```diff
@@ -161,13 +166,20 @@ def isPendingInstall(self) -> bool:
 			# have not been installed yet
 			addonDataManager._downloadsPendingInstall,
 		)
+		nameInDownloadsPendingCompletion = filter(
+			lambda m: m.model.name == self.name,
+			# add-ons which are currently being downloaded
+			# and have not been cancelled
+			addonDataManager._downloadsPendingCompletion,
+		)
 		return (
 			super().isPendingInstall
 			# True if this add-on has been downloaded but
 			# has not been installed yet
 			or bool(next(nameInDownloadsPendingInstall, False))
 			# True if this add-on is currently being downloaded
-			or os.path.exists(self.tempDownloadPath)
+			# and the download has not been cancelled
+			and bool(next(nameInDownloadsPendingCompletion, False))
```

Comment:
have you tested this?
I think it needs to be
```suggestion
			# and the download has not been cancelled
			or bool(next(nameInDownloadsPendingCompletion, False))
```

Reply:
I apologise for the oversight. I applied the suggestion directly.

Reply:
yep my mistake there too